src/app/components/views/drawer/drawer.component.ts

```diff
@@ -0,0 +1,47 @@
+import { Component, OnInit, OnDestroy, CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
+import { DrawerService } from '../../../services/drawer.service';
+import { NgIcon } from '@ng-icons/core';
+import { NgIf } from '@angular/common';
+import { CommonModule } from '@angular/common';
+import { Subscription } from 'rxjs';
+import '@ulb-darmstadt/shacl-form';
+
+@Component({
+  selector: 'app-drawer',
+  standalone: true,
+  imports: [NgIcon, NgIf, CommonModule],
+  templateUrl: './drawer.component.html',
+  styleUrls: ['./drawer.component.scss'],
+  schemas: [CUSTOM_ELEMENTS_SCHEMA]
+})
+export class DrawerComponent implements OnInit, OnDestroy {
+  drawerItems: string[] = [];
+  drawerSubscription: Subscription = new Subscription();
+
+  constructor(private drawerService: DrawerService) { }
+
+  ngOnInit(): void {
+    this.drawerSubscription = this.drawerService.drawerItems$.subscribe(items => {
+      if (items.length > 0) {
+        this.drawerItems = items;
+        this.openDrawer();
+      }
+    });
+  }
+
+  ngOnDestroy(): void {
+    if (this.drawerSubscription) {
+      this.drawerSubscription.unsubscribe();
+    }
+  }
+
+  openDrawer(): void {
+    const drawerCheckbox = document.getElementById('my-drawer-4') as HTMLInputElement;
```

Comment:
In Angular, you typically do not get elements by ID from the document, but instead use ViewChild. https://github.com/Simon-Dirks/view-a-LOD/tree/main/src/app/components/node/node-render-components/predicate-render-components/map-thumb gives an example implementation. Note that the element is only initialized after the "AfterViewInit" lifecycle hook, so you might need to check if the ElementRef is undefined to be sure it's properly initialized when you run your function. See also https://angular.dev/guide/components/lifecycle

Reply:
Alternatively, you can also bind the [checked] attribute of the checkbox in the html to a variable (e.g., "opened"), and then set that variable in that function. That might make more sense than directly manipulating the DOM.

Reply:
Changed it to data binding (your second option)